### PR TITLE
Improve progress bar

### PR DESCRIPTION
**Pull request by xibr**
#147

### CODE REVIEW

File: library/src/main/java/com/yausername/youtubedl_android/DownloadProgressCallback.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 package com.yausername.youtubedl_android;
 
 public interface DownloadProgressCallback {
-    void onProgressUpdate(float progress, long etaInSeconds, String line);
+    void onProgressUpdate(String line);
```
Comment (yausername) [inline library/src/main/java/com/yausername/youtubedl_android/DownloadProgressCallback.java:4]: Can we do this change without breaking the API?
I mean keep the same method signature as before.
We can use some reasonable defaults for `progress` and `etaInSeconds` when the line does not match the download pattern. For example 0, -1 before the download starts and 100, 0 after it is completed.

File: library/src/main/java/com/yausername/youtubedl_android/utils/Utils.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.yausername.youtubedl_android.utils;
+
+import androidx.annotation.NonNull;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class Utils {
```
Comment (yausername) [inline library/src/main/java/com/yausername/youtubedl_android/utils/Utils.java:8]: Won't need this if we stick to the old function signature for the callback.

File: library/src/main/java/com/yausername/youtubedl_android/DownloadProgressCallback.java (hunk 1 above)
Comment (xibr) [reply library/src/main/java/com/yausername/youtubedl_android/DownloadProgressCallback.java]: Done, maybe it's better now.

File: library/src/main/java/com/yausername/youtubedl_android/StreamProcessExtractor.java [hunk 3]
```diff
@@ -11,6 +11,11 @@
 import java.util.regex.Pattern;
 
 class StreamProcessExtractor extends Thread {
+
+    private static final String TAG = StreamProcessExtractor.class.getSimpleName();
+
+    private static final long ETA = 0;
```
Comment (yausername) [inline library/src/main/java/com/yausername/youtubedl_android/StreamProcessExtractor.java:17]: nit: how about `-1` and `-1.0f` to indicate unknown values.

**Review by yausername — APPROVED** [review]
Looks good. Left another nit comment.

**Pull request merged**